Common/Data/Custom/SmartInsider/SmartInsiderIntention.cs

```diff
@@ -0,0 +1,339 @@
+                CompanyID = CompanyID,
+                ICBIndustry = ICBIndustry,
+                ICBSuperSector = ICBSuperSector,
+                ICBSector = ICBSector,
+                ICBSubSector = ICBSubSector,
+                ICBCode = ICBCode,
+                CompanyName = CompanyName,
+                PreviousResultsAnnouncementDate = PreviousResultsAnnouncementDate,
+                NextResultsAnnouncementsDate = NextResultsAnnouncementsDate,
+                NextCloseBegin = NextCloseBegin,
+                LastCloseEnded = LastCloseEnded,
+                SecurityDescription = SecurityDescription,
+                TickerCountry = TickerCountry,
+                TickerSymbol = TickerSymbol,
+                AnnouncementDate = AnnouncementDate,
+                TimeReleased = TimeReleased,
+                TimeProcessed = TimeProcessed,
+                TimeReleasedUtc = TimeReleasedUtc,
+                TimeProcessedUtc = TimeProcessedUtc,
+                AnnouncedIn = AnnouncedIn,
+
+                IntentionVia = IntentionVia,
+                IntentionBy = IntentionBy,
+                BuybackIntentionHoldingType = BuybackIntentionHoldingType,
+                IntentionAmount = IntentionAmount,
+                ValueCurrency = ValueCurrency,
+                IntentionValue = IntentionValue,
+                IntentionPercentage = IntentionPercentage,
+                IntentionAuthorisationStartDate = IntentionAuthorisationStartDate,
+                IntentionAuthorisationEndDate = IntentionAuthorisationEndDate,
+                PriceCurrency = PriceCurrency,
+                MinimumPrice = MinimumPrice,
+                MaximumPrice = MaximumPrice,
+                BuybackIntentionNoteText = BuybackIntentionNoteText
+            };
+        }
+
+        /// <summary>
+        /// Converts the data to CSV
+        /// </summary>
+        /// <returns>String of CSV</returns>
+        /// <remarks>Parsable by the constructor should you need to recreate the object from CSV</remarks>
+        public override string ToCsv()
+        {
+            return $"{TransactionID}\t{BuybackType}\t{LastUpdate:yyyyMMdd}\t{LastIDsUpdate:yyyyMMdd}\t{ISIN}\t{USDMarketCap}\t{CompanyID}\t{ICBIndustry}\t{ICBSuperSector}\t{ICBSector}\t{ICBSubSector}\t{ICBCode}\t{CompanyName}\t{PreviousResultsAnnouncementDate:yyyyMMdd}\t{NextResultsAnnouncementsDate:yyyyMMdd}\t{NextCloseBegin:yyyyMMdd}\t{LastCloseEnded:yyyyMMdd}\t{SecurityDescription}\t{TickerCountry}\t{TickerSymbol}\t{AnnouncementDate:yyyyMMdd}\t{TimeReleased:yyyyMMdd HH:mm:ss}\t{TimeProcessed:yyyyMMdd HH:mm:ss}\t{TimeReleasedUtc:yyyyMMdd HH:mm:ss}\t{TimeProcessedUtc:yyyyMMdd HH:mm:ss}\t{AnnouncedIn}\t{IntentionVia}\t{IntentionBy}\t{BuybackIntentionHoldingType}\t{IntentionAmount}\t{ValueCurrency}\t{IntentionValue}\t{IntentionPercentage}\t{IntentionAuthorisationStartDate:yyyyMMdd}\t{IntentionAuthorisationEndDate:yyyyMMdd}\t{PriceCurrency}\t{MinimumPrice}\t{MaximumPrice}\t{BuybackIntentionNoteText}";
```

Comment:
The method claims `ToCsv` but this is generating a `.tsv` (tab-separated values` -- _also_ when dealing with a long list like this, recommend putting each item on it's  own line. In general we shouldn't have lines of code over ~150 characters (usually limited to ~120 characters w/ some exceptions).

If this is intended as tab separated, please change the method name and documentation, otherwise, update the implementation. Also, please use a construct such as `string.Join`:
``` csharp
return string.Join("\t",
    TransactionID,
    BuybackType,
    LastUpdate.ToString("yyyyMMdd"),
    // ....
);
```

Reply:
Yes, I considered using the TSV extension, but ultimately decided against it since I wanted to keep the extensions consistent with the rest of LEAN. 

And yes, for formatting I should definitely be using `string.Join` 👍